core/src/Ledger.sol

```diff
@@ -170,17 +170,15 @@ contract Ledger {
 
     function underlyingBalance(address account) external view returns (uint256) {
         (, uint256 inventory, uint256 newTotalSupply) = _previewInterest(_getCache());
-        return _nominalAssets(account, inventory, newTotalSupply);
+        return _convertToAssets(_nominalShares(account, inventory, newTotalSupply), inventory, newTotalSupply, false);
     }
 
     function underlyingBalanceStored(address account) external view returns (uint256) {
         unchecked {
-            return
-                _nominalAssets({
-                    account: account,
-                    inventory: lastBalance + (uint256(borrowBase) * borrowIndex) / BORROWS_SCALER,
-                    totalSupply_: totalSupply
-                });
+            uint256 inventory = lastBalance + (uint256(borrowBase) * borrowIndex) / BORROWS_SCALER;
+            uint256 totalSupply_ = totalSupply;
+
+            return _convertToAssets(_nominalShares(account, inventory, totalSupply_), inventory, totalSupply_, false);
```

Comment:
I'm sure this works, but why not pass `totalSupply` directly?

Reply:
We use `totalSupply_` twice on this line, so it's better to cache it in memory than to read from storage twice.